src/tests/ftest/util/host_utils.py

```
@@ -0,0 +1,338 @@
+"""
+(C) Copyright 2018-2022 Intel Corporation.
+
+SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import os
+import re
+
+from ClusterShell.NodeSet import NodeSet, NodeSetParseError
+
+from run_utils import RunException, run_local, get_local_host
+
+
+class HostException(Exception):
+    """Base exception for this module."""
+
+
+def get_host_parameters(test, hosts_key, partition_key, reservation_key, namespace):
+    """Get the host parameters from the test yaml file.
+
+    Args:
+        test (Test): the avocado test class
+        hosts_key (str): test yaml key used to obtain the set of hosts to test
+        partition_key (str): test yaml key used to obtain the host partition name
+        reservation_key (str): test yaml key used to obtain the host reservation name
+        namespace (str): test yaml path to the keys
+
+    Returns:
+        (object, str, str): the hosts, partition, and reservation obtained form the test yaml
+
+    """
+    reservation_default = os.environ.get("_".join(["DAOS", reservation_key.upper()]), None)
+
+    # Collect any host information from the test yaml
+    hosts = test.params.get(hosts_key, namespace)
+    partition = test.params.get(partition_key, namespace)
+    reservation = test.params.get(reservation_key, namespace, reservation_default)
+    if partition is not None and hosts is not None:
+        test.fail(
+            f"Specifying both a '{partition_key}' partition and '{hosts_key}' set of hosts is not "
+            "supported!")
```

Would it be better to raise an exception instead of calling fail directly?